Tests/Realm.Tests/Sync/SynchronizedInstanceTests.cs

```diff
@@ -396,6 +396,89 @@ public void EmbeddedObject_WhenAdditiveExplicit_ShouldThrow()
             });
         }
 
+        [Test]
+        public void WriteCopy_LocalToSync([Values(true, false)] bool originalEncrypted,
+                                          [Values(true, false)] bool copyEncrypted)
+        {
+            SyncTestHelpers.RunBaasTestAsync(async () =>
+            {
+                var originalConfig = new RealmConfiguration(Guid.NewGuid().ToString())
+                {
+                    Schema = new[] { typeof(ObjectIdPrimaryKeyWithValueObject) }
+                };
+                if (originalEncrypted)
+                {
+                    originalConfig.EncryptionKey = TestHelpers.GetEncryptionKey(42);
+                }
+
+                var copyConfig = await GetIntegrationConfigAsync(Guid.NewGuid().ToString());
+                if (copyEncrypted)
+                {
+                    copyConfig.EncryptionKey = TestHelpers.GetEncryptionKey(23);
+                }
+
+                File.Delete(copyConfig.DatabasePath);
+
+                using var originalRealm = GetRealm(originalConfig);
+
+                AddDummyData(originalRealm, true);
+
+                originalRealm.WriteCopy(copyConfig);
+
+                using var copiedRealm = GetRealm(copyConfig);
+
+                Assert.AreEqual(copiedRealm.All<ObjectIdPrimaryKeyWithValueObject>().Count(), DummyDataSize / 2);
+
+                var fromCopy = copiedRealm.Write(() =>
+                {
+                    return copiedRealm.Add(new ObjectIdPrimaryKeyWithValueObject
+                    {
+                        StringValue = "Added from copy"
+                    });
+                });
+
+                // We cannot re-sync like we do it in `WriteCopy_CanSynchronizeData` since the `originalRealm` is not synced but instead
+                // at least try to upload and download the data in the `copiedRealm`.
+                await WaitForUploadAsync(copiedRealm);
+                await WaitForDownloadAsync(copiedRealm);
```

Comment:
That was my idea, yeah. Since only one of the two Realms is synced I can't test a full sync cycle like I did in the other test where we copy a synced Realm to a synced Realm but I wanted to test as much as possible and figured I could at least try to call upload and download to get soooome kind of indication if something completely went wrong.

Do you have any other ideas what would be good to check here?

Reply:
I think that we already test that sync works in other contexts, so probably we don't need to do this here too in my opinion. Checking that the synced realm has the same kind of objects as the original one should be enough

Reply:
[af29ecc](https://github.com/realm/realm-dotnet/pull/2900/commits/af29eccefda22e78cae23bac6b3381643d068094)